### PR TITLE
Add package geos v3.11.1

### PR DESCRIPTION
**Pull request by sindhu-karri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Adding geos package, as it is required to build GDAL (as requested by customer - Azure farmbeats)

###### Change Log  

- Initial CBL-Mariner import from Fedora 38 (license: MIT)
- Removed building for mingw environment
- License verified

###### Does this affect the toolchain?  

NO

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: AMD buddy build
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=317551&view=results
- ARM64 buddy build
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=317555&view=results

### CODE REVIEW

File: SPECS/geos/geos.spec [hunk 1]
```diff
@@ -0,0 +1,385 @@
+Summary:        GEOS is a C++ port of the Java Topology Suite
+Name:           geos
+Version:        3.11.1
+Release:        4%{?dist}
+License:        LGPLv2
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://trac.osgeo.org/geos/
+Source0:        http://download.osgeo.org/%{name}/%{name}-%{version}.tar.bz2
```
Comment (Sumynwa) [inline SPECS/geos/geos.spec:9]: Kindly also use https for Source.

File: SPECS/geos/geos.spec [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+%cmake_build
+
+
+%install
+%cmake_install
+make docs -C %{__cmake_builddir}
+
+
+%check
+%ifnarch s390x
```
Comment (Sumynwa) [inline SPECS/geos/geos.spec:55]: We can remove reference to s390x as this arch is not supported.

File: SPECS/geos/geos.spec [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+JTS in C++. This includes all the OpenGIS "Simple Features for SQL" spatial
+predicate functions and spatial operators, as well as specific JTS topology
+functions such as IsValid().
+
+This package contains the development files to build applications that
+use GEOS.
+
+%prep
+%autosetup -p1
+
```
Comment (Sumynwa) [inline SPECS/geos/geos.spec:41]: Just nit-pick Remove extra blank lines.

**Review by Sumynwa — CHANGES_REQUESTED** [review]

File: SPECS/geos/geos.spec [hunk 4]
```diff
@@ -0,0 +1,385 @@
+Summary:        GEOS is a C++ port of the Java Topology Suite
+Name:           geos
+Version:        3.11.1
```
Comment (0xba1a) [inline SPECS/geos/geos.spec:3]: Is this the latest version? If not at least check whether this version has any CVEs. We shouldn't introduce any package with CVEs existing.

File: SPECS/geos/geos.spec [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+%{_libdir}/libgeos_c.so
+%{_libdir}/libgeos.so
+%{_libdir}/cmake/GEOS/
+%{_libdir}/pkgconfig/%{name}.pc
+
+%changelog
+* Tue Feb 14 2023 Sindhu Karri <lakarri@microsoft.com> - 3.11.1-4
+- Initial CBL-Mariner import from Fedora 38 (license: MIT)
+- Removed building for mingw environment
+- License verified
```
Comment (0xba1a) [inline SPECS/geos/geos.spec:82]: 👍

File: SPECS/geos/geos.spec (hunk 4 above)
Comment (sindhu-karri) [reply SPECS/geos/geos.spec:3]: This is the latest version.

**Review by Sumynwa — APPROVED** [review]
LGTM

**Review by 0xba1a — APPROVED** [review]

**Pull request merged**